shared/macros-ansible.jinja

```diff
@@ -0,0 +1,128 @@
+{{{ ansible_lineinfile("Insert correct line into " + path, path, regex=None, new_line=new_line, create=create, state='present')|indent }}}
+{{%- endif %}}
+{{%- endmacro %}}
+
+{{#
+  Ensure the configuration is set in a file. Note this handles generic
+  key-seperator-value files with no sense of structure. In particular,
+  ini configuration files are best served with the ini Ansible module
+  instead of lineinfile-based solutions.
+#}}
+{{%- macro ansible_set_config_file(msg, file, parameter, separator=' ', separator_regex='\s+', value='', prefix_regex='^\s*', create='no', validate='') %}}
+{{{ ansible_only_lineinfile(msg, file, prefix_regex + parameter + separator_regex, parameter + separator + value, create=create, block=True, validate=validate) }}}
+{{%- endmacro %}}
+
+{{#
+  Ensure the configuration is set in a file and not conflicted by a
+  configuration in a directory. Note this handles generic key-separator-value
+  files with no sense of structure. In particular, ini configuration files are
+  best served with the ini Ansible module instead of lineinfile-based
+  solutions.
+#}}
+{{%- macro ansible_set_config_file_dir(msg, config_file, config_dir, set_file, parameter, separator=' ', separator_regex='\s+', value='', prefix_regex='^\s*', create='no', validate='') %}}
+{{%- set var_dir = config_dir | replace("/", "_") | replace("-", "_") | replace(".", "_") -%}}
+{{%- set dir_exists = var_dir + "_exists" -%}}
+{{%- set dir_parameter = var_dir + "_has_parameter" -%}}
+{{%- set line_regex = prefix_regex + parameter + separator_regex -%}}
+{{%- set find_when = dir_exists + ".stat.isdir is defined and " + dir_exists + ".stat.isdir" -%}}
+{{%- set lineinfile_items = "{{ " + dir_parameter + ".files }}" -%}}
+{{%- set lineinfile_when = dir_parameter + ".matched" -%}}
+{{%- set new_line = parameter + separator + value -%}}
+- name: '{{{ msg or rule_title }}}'
+  block:
+    {{{ ansible_lineinfile("Deduplicate values from " + config_file, config_file, regex=line_regex, create='no', state='absent')|indent }}}
+    {{{ ansible_stat("Check if " + config_dir + " exists", path=config_dir, register=dir_exists)|indent }}}
+    {{{ ansible_find("Check if the parameter " + parameter + " is present in " + config_dir, paths=config_dir, contains=line_regex, register=dir_parameter, when=find_when)|indent }}}
+    {{{ ansible_lineinfile("Remove parameter from files in " + config_dir, path="{{ item.path }}", regex=line_regex, state="absent", with_items=lineinfile_items, when=lineinfile_when)|indent }}}
+    {{{ ansible_lineinfile("Insert correct line to " + set_file, set_file, new_line=new_line, create=create, state='present', validate=validate)|indent }}}
+{{%- endmacro %}}
+
+{{%- macro ansible_sshd_set(msg, parameter, value) %}}
+{{{ ansible_set_config_file(msg, "/etc/ssh/sshd_config", parameter, value=value, create="yes", prefix_regex='(?i)^\s*', validate="/usr/sbin/sshd -t -f %s") }}}
```

Comment:
`sshd` configuration (unlike `ssh` client configuration) doesn't have a directory. Parameters are case sensitive, but values aren't. We also have a utility to validate the new configuration.